src/java.base/share/classes/jdk/internal/foreign/abi/aarch64/macos/MacOsAArch64VaList.java

```diff
@@ -170,12 +165,12 @@ public MemoryAddress address() {
 
     public static non-sealed class Builder implements VaList.Builder {
 
-        private final MemorySession session;
+        private final MemorySessionImpl session;
         private final List<SimpleVaArg> args = new ArrayList<>();
 
         public Builder(MemorySession session) {
-            MemorySessionImpl.toSessionImpl(session).checkValidStateSlow();
-            this.session = session;
+            ((MemorySessionImpl)session).checkValidState();
```

Comment:
Or here, if the memory session is a non-closeable view.

Reply:
I believe there was a wrong renaming with the IDE here, I will fix this